[TECH] Suppression de la mise en cache sur certaines routes pour gérer le cache par Baleen

Mefl opened the pull request:
## :pancakes: Problème
Certaines routes sont quasiment statiques mais sont servies à chaque fois car baleen ne les cache pas à cause du header cache-control

## :bacon: Proposition

supprimer le header cache-control

## Remarque
Depuis Hapi.dev v17, config a été déprécié et remplacé par options. Toutefois le nom config restait pour compatibilité ascendante.
Or lors de nos tests, il semble que config.cache ne soit pas pris en compte, d'où le changement vers la notation state of the art afin de pouvoir changer le comportement du cache.

Mefl committed:
remove cache header

pix-bot-github commented:
Une fois les applications déployées, elles seront accessibles via les liens suivants :
  * [App (.fr)](https://app-pr11189.review.pix.fr)
  * [App (.org)](https://app-pr11189.review.pix.org)
  * [Orga (.fr)](https://orga-pr11189.review.pix.fr)
  * [Orga (.org)](https://orga-pr11189.review.pix.org)
  * [Certif (.fr)](https://certif-pr11189.review.pix.fr)
  * [Certif (.org)](https://certif-pr11189.review.pix.org)
  * [Junior](https://junior-pr11189.review.pix.fr)
  * [Admin](https://admin-pr11189.review.pix.fr)
  * [API](https://api-pr11189.review.pix.fr/api/)
  * [Audit Logger](https://pix-audit-logger-review-pr11189.osc-fr1.scalingo.io/api/)

Les variables d'environnement seront accessibles via les liens suivants :
  * [scalingo front](https://dashboard.scalingo.com/apps/osc-fr1/pix-front-review-pr11189/environment)
  * [scalingo api](https://dashboard.scalingo.com/apps/osc-fr1/pix-api-review-pr11189/environment)
  * [scalingo audit-logger](https://dashboard.scalingo.com/apps/osc-fr1/pix-audit-logger-review-pr11189/environment)

Mefl committed:
Modification du config en options pour être conforme au comportement de Hapi.dev v17+

config étant gardé pour compatibilité ascendante mais ne permet pas de prendre en compte les nouveaux comportement comme la spéiification du comportement de header de cache.

lego-technix commented on api/src/identity-access-management/application/oidc-provider/oidc-provider.route.js:
```diff
@@ -7,13 +7,14 @@ export const oidcProviderRoutes = [
     method: 'GET',
     path: '/api/oidc/identity-providers',
-    config: {
+    options: {
```
Ce renommage ne correspond pas au titre du commit.

Il serait préférable : 
* soit de ne pas faire cette modification du tout
* soit de faire cette modification dans un autre commit et d'expliquer pourquoi

Mefl replied:
Depuis Hapi.dev v17, config a été déprécié et remplacé par options. Toutefois le nom config restait pour compatibilité ascendante.
Or lors de nos tests, il semble que config.cache ne soit pas pris en compte, d'où le changement vers la notation state of the art afin de pouvoir changer le comportement du cache.

lego-technix replied:
Bien reçu. Donc faire un commit dédié et qui explique de manière minimaliste pourquoi STP. Et mettre l'explication que tu viens d'écrire dans la description de la PR STP :pray:

Mefl replied:
done

Mefl committed:
Désactivation de la gestion du header de cache afin que la ressource puisse être cachée par Baleen

lego-technix reviewed (commented):
Bizarre, lorsque je teste en local le header `cache-control` est bien différent d'avant cette PR, mais il est toujours présent :thinking: 

Avant : 
```shell
curl --insecure --include https://app.dev.pix.fr/api/feature-toggles
HTTP/1.1 200 OK
Server: nginx
Date: Mon, 27 Jan 2025 15:49:32 GMT
Content-Type: application/json; charset=utf-8
Content-Length: 782
Connection: keep-alive
X-Powered-By: Express
cache-control: max-age=30, must-revalidate
Vary: origin, Accept-Encoding
access-control-expose-headers: WWW-Authenticate,Server-Authorization
```

Avec cette PR : 
```shell
curl --insecure --include https://app.dev.pix.fr/api/feature-toggles
HTTP/1.1 200 OK
Server: nginx
Date: Mon, 27 Jan 2025 15:50:02 GMT
Content-Type: application/json; charset=utf-8
Content-Length: 782
Connection: keep-alive
X-Powered-By: Express
Cache-Control: private, max-age=0, must-revalidate
Vary: origin, Accept-Encoding
access-control-expose-headers: WWW-Authenticate,Server-Authorization
```

lego-technix reviewed (approved):
✅ Lu et testé fonctionnellement avec succès

Sur les RA, comme prévu, les routes modifiées n'ont bien pas de header `cache-control`.
